src/hooks/use-mutation/useMutation.ts

```diff
@@ -0,0 +1,44 @@
+import { useCallback, useState } from 'react';
+import { TMutationFunction, IUseMutation, TStatus, TMutationParams } from './useMutation.types';
+import { AxiosError, AxiosResponse } from 'axios';
+import { axiosInstance } from 'services';
+
+export const useMutation = (): IUseMutation => {
+  const [data, setData] = useState<AxiosResponse | null>(null);
+  const [error, setError] = useState<AxiosError | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isError, setIsError] = useState<boolean>(false);
+  const [isSuccess, setIsSuccess] = useState<boolean>(false);
+  const [status, setStatus] = useState<TStatus>('idle');
+
+  const mutate: TMutationFunction = useCallback(
+    async ({ data, url, method = 'post', onError, onSuccess }: TMutationParams): Promise<void> => {
```

Comment:
No need to add type for function params. It is already added in TMutationFunction